playground-env/src/main/scala/org/ergoplatform/playgroundenv/utils/ErgoScriptCompiler.scala

```diff
@@ -24,4 +26,36 @@ object ErgoScriptCompiler {
     val prop = compiler.compile(liftedEnv, ergoScript)
     ErgoContract(_ => ???, prop)
   }
+
+  def printErgoScriptEncodingsAndHashesFromUrl(url: String) = {
+    val script           = scala.io.Source.fromURL(url).mkString
+    val ergoTree         = compile(Map(), script).ergoTree
+    implicit var encoder = new ErgoAddressEncoder(MainnetNetworkPrefix)
+    var addr             = Pay2SAddress(ergoTree)
+
+    val addrTypePrefix = Pay2SAddress.addressTypePrefix
+    val mainnetEncoded = encoder.toString(addr)
+    val mainnetHash = Base58.encode(
+      ErgoAddressEncoder
+        .hash256((MainnetNetworkPrefix + addrTypePrefix).toByte +: ergoTree.bytes)
+    )
+
+    encoder = new ErgoAddressEncoder(TestnetNetworkPrefix)
+    addr    = Pay2SAddress(ergoTree)
+    val testnetEncoded = encoder.toString(addr)
+    val testnetHash = Base58.encode(
+      ErgoAddressEncoder
+        .hash256((TestnetNetworkPrefix + addrTypePrefix).toByte +: ergoTree.bytes)
+    )
+
+    val serializedErgoTree = Base16.encode(ergoTree.bytes)
+    println(
+      s"Ergoscript @$url\n" +
+      s"P2S(mainnet): $mainnetEncoded\n" +
+      s"P2S(mainnet hash): $mainnetHash\n" +
```

Comment:
We should emphasize that the hash has nothing to do with P2S. The last thing we want is for someone to try to use this hash instead of the address. I wonder if using base64 might be a better fit to avoid similarity with addresses.

Reply:
I've done away entirely with hashes of P2S, and used base64 encoding for hash of ergo-tree.